hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/tagdetails/TagAssignementComboBox.java

```diff
@@ -0,0 +1,165 @@
+import org.eclipse.hawkbit.ui.common.tagdetails.TagPanelLayout.TagAssignmentListener;
+import org.eclipse.hawkbit.ui.utils.SPUIStyleDefinitions;
+import org.eclipse.hawkbit.ui.utils.UIComponentIdProvider;
+import org.eclipse.hawkbit.ui.utils.UIMessageIdProvider;
+import org.eclipse.hawkbit.ui.utils.VaadinMessageSource;
+
+import com.google.common.collect.Sets;
+import com.vaadin.data.Item;
+import com.vaadin.data.util.IndexedContainer;
+import com.vaadin.ui.ComboBox;
+import com.vaadin.ui.HorizontalLayout;
+import com.vaadin.ui.themes.ValoTheme;
+
+/**
+ * Combobox that lists all available Tags that can be assigned to a
+ * {@link Target} or {@link DistributionSet}.
+ */
+public class TagAssignementComboBox extends HorizontalLayout {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final String NAME_PROPERTY = "name";
+    private static final String COLOR_PROPERTY = "color";
+
+    private final IndexedContainer allAssignableTags;
+    private final transient Set<TagAssignmentListener> listeners = Sets.newConcurrentHashSet();
+
+    private final ComboBox assignableTagsComboBox;
+
+    private final boolean readOnlyMode;
+
+    /**
+     * Constructor.
+     * 
+     * @param i18n
+     *            the i18n
+     * @param readOnlyMode
+     *            if true the combobox will be disabled so no assignment can be
+     *            done.
+     */
+    TagAssignementComboBox(final VaadinMessageSource i18n, final boolean readOnlyMode) {
+
+        this.readOnlyMode = readOnlyMode;
+
+        setWidth("100%");
+
+        allAssignableTags = new IndexedContainer();
+        allAssignableTags.addContainerProperty(NAME_PROPERTY, String.class, "");
+        allAssignableTags.addContainerProperty(COLOR_PROPERTY, String.class, "");
+
+        assignableTagsComboBox = new ComboBoxBuilder().setId(UIComponentIdProvider.TAG_SELECTION_ID)
+                .setPrompt(i18n.getMessage(UIMessageIdProvider.TOOLTIP_SELECT_TAG))
+                .setValueChangeListener(e -> onSelectionChanged()).buildCombBox();
+        addComponent(assignableTagsComboBox);
+        assignableTagsComboBox.setContainerDataSource(allAssignableTags);
+        assignableTagsComboBox.setNullSelectionAllowed(true);
+        assignableTagsComboBox.select(assignableTagsComboBox.getNullSelectionItemId());
+        assignableTagsComboBox.addStyleName(SPUIStyleDefinitions.DETAILS_LAYOUT_STYLE);
+        assignableTagsComboBox.addStyleName(ValoTheme.COMBOBOX_TINY);
+        assignableTagsComboBox.setEnabled(!readOnlyMode);
+        assignableTagsComboBox.setWidth("100%");
+    }
+
+    /**
+     * Initializes the Combobox with all assignable tags.
+     * 
+     * @param assignableTags
+     *            assignable tags
+     */
+    void initializeAssignableTags(final List<TagData> assignableTags) {
+        assignableTags.forEach(this::addAssignableTag);
+    }
+
+    private void onSelectionChanged() {
```

Comment:
change the method signature to accept event, so you could use method reference in constructor and get the selected value from the event